src/Compilers/Core/Portable/Emit/EditAndContinue/DeletedMethodDefinition.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis.Emit.EditAndContinue
+{
+    internal sealed partial class DeletedMethodDefinition : IMethodDefinition
+    {
+        private IMethodSymbolInternal _methodDef;
+        private IMethodDefinition _oldMethod;
+
+        public DeletedMethodDefinition(IMethodDefinition oldMethod, IMethodSymbolInternal methodDef)
+        {
+            _oldMethod = oldMethod;
+            _methodDef = methodDef;
+        }
+
+        public IEnumerable<IGenericMethodParameter> GenericParameters => _oldMethod.GenericParameters;
```

Comment:
>_oldMethod.GenericParameters [](http://example.com/codeflow?start=73&length=28)

This method should have its own type parameters. Wrappers are fine, but owned by this method rather than by the old one.